pvlib/iotools/nasa_power.py

```diff
@@ -0,0 +1,153 @@
+"""Functions for reading and retrieving data from NASA POWER."""
+
+import pandas as pd
+import requests
+import numpy as np
+
+URL = 'https://power.larc.nasa.gov/api/temporal/hourly/point'
+
+DEFAULT_PARAMETERS = [
+    'ALLSKY_SFC_SW_DNI', 'ALLSKY_SFC_SW_DIFF', 'ALLSKY_SFC_SW_DWN',
+    'T2M', 'WS10M'
+]
+
+VARIABLE_MAP = {
+    'ALLSKY_SFC_SW_DWN': 'ghi',
+    'ALLSKY_SFC_SW_DIFF': 'dhi',
+    'ALLSKY_SFC_SW_DNI': 'dni',
+    'CLRSKY_SFC_SW_DWN': 'ghi_clear',
+    'T2M': 'temp_air',
+    'WS2M': 'wind_speed_2m',
+    'WS10M': 'wind_speed',
+}
+
+
+def get_nasa_power(latitude, longitude, start, end,
+                   parameters=DEFAULT_PARAMETERS, community='re', url=URL,
+                   elevation=None, wind_height=None, wind_surface=None,
+                   map_variables=True):
+    """
+    Retrieve irradiance and weather data from NASA POWER.
+
+    A general description of NASA POWER is given in [1]_ and the API is
+    described in [2]_. A detailed list of the available parameters can be
+    found in [3]_.
+
+    Parameters
+    ----------
+    latitude: float
+        In decimal degrees, north is positive (ISO 19115).
+    longitude: float
+        In decimal degrees, east is positive (ISO 19115).
+    start: datetime like
+        First timestamp of the requested period.
+    end: datetime like
+        Last timestamp of the requested period.
+    parameters: str, list
+        List of parameters. The default parameters are mentioned below; for the
+        full list see [3]_. Note that the pvlib naming conventions can also be
+        used.
+
+        * ``ALLSKY_SFC_SW_DWN``: Global Horizontal Irradiance (GHI) [Wm⁻²]
+        * ``ALLSKY_SFC_SW_DIFF``: Diffuse Horizontal Irradiance (DHI) [Wm⁻²]
+        * ``ALLSKY_SFC_SW_DNI``: Direct Normal Irradiance (DNI) [Wm⁻²]
+        * ``T2M``: Air temperature at 2 m [C]
+        * ``WS10M``: Wind speed at 10 m [m/s]
```

Comment:
I suggest using pvlib names here and in the default parameter list.  As a user, I don't want to have to mentally translate `ALLSKY_SFC_SW_DWN` into `ghi`.

Reply:
The pvlib function currently accepts both of those and I am not suggesting changing that.  The suggestion is for the default values to use pvlib names instead of NASA POWER names, to (IMHO) make the docstring a little clearer.